tests/e2e/e2e.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+# Copyright 2019 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eux
+
+export ARTIFACTS="${ARTIFACTS:-$(mktemp -d)}"
+
+function cleanup_kind_cluster() {
+    kind export logs --name istio-testing "${ARTIFACTS}/kind"
+#    kind delete cluster --name=istio-testing
+}
+
+function setup_kind_cluster() {
+  # Delete any previous e2e KinD cluster
+  echo "Deleting previous KinD cluster with name=istio-testing"
+  if ! (kind delete cluster --name=istio-testing) > /dev/null; then
+    echo "No existing kind cluster with name istio-testing. Continue..."
+  fi
+
+  trap cleanup_kind_cluster EXIT
+
+  # Create KinD cluster
+  if ! (kind create cluster --name=istio-testing --loglevel debug --retain --wait 30s); then
+    echo "Could not setup KinD environment. Something wrong with KinD setup. Exporting logs."
+    exit 1
+  fi
+
+  KUBECONFIG="$(kind get kubeconfig-path --name="istio-testing")"
+  export KUBECONFIG
+
+cat <<EOF > metallb-config.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: metallb-system
+  name: config
+data:
+  config: |
+    address-pools:
+    - name: default
+      protocol: layer2
+      addresses:
+      - 172.17.255.1-172.17.255.250
```

Comment:
will this work in all environments? An alternative used in installer is a nodeport

Reply:
I am not sure.  It works where we need it the most - which is automated testing.  I have read in the past this may not work well on mac, but I have no reason to think it may not.  I'd suggest a little latitude here, so I can prove this out, and if things are not working, I can setup the call to --use-local or whatever it is for e2e.  We have a project goal of getting rid of nodeport testing...